### DISCUSSION TITLE
feat: Automatically compress images

### DISCUSSION THREAD

**Discussion by Algorithm5838** [open-webui/open-webui#3395] [Ideas]
**Is your feature request related to a problem? Please describe.**
Large image sizes increase response time and can exceed payload size limits.

**Describe the solution you'd like**
An option to automatically compress images upon upload or generation.

**Describe alternatives you've considered**
Manually compressing images before use is cumbersome.

**Additional context**
This feature would improve user experience and reduce errors related to large image sizes.

**Comment by rwjack** [comment]
Yeah, the latest cuda image is 8.2G. Wack

**Comment by relink2013** [comment]
I came here specifically to ask this. I didn't realize that the size of the image was going to play a factor, to be honest I assumed it was already downscaling automatically. 

Throw ImageMagick in in upload process and thats all that would really be needed. Then add a setting in the admin panel to allow specifying a resolution OR a size in MB. I have run across some models that specify max resolutions, and other that specify a max file size, so having both just makes things a little less confusing on the users end.

**Reply by rwjack** [reply]
I think this is about the openwebui image itself, not the model images downloaded by ollama?

**Reply by Algorithm5838** [reply]
To clarify - I'm referring to picture/photo files, not Docker images. When working with vision model, I've noticed that images larger than 1MB take significant time to upload. Additionally, OpenRouter's API has a 5MB payload limit, so images exceeding this size are automatically rejected. I've previously used another project that automatically compressed images, which resulted in instant model responses.

**Reply by rwjack** [reply]
Ohhhhh, this really is about images then 😅
MB, I'll open up another one

**Reply by relink2013** [reply]
Yeah sorry, I mean photo images. Understandable mix up though, tbh I had to read the issue about three times to make sure I wasn't about to do the same.

**Comment by prasoon2211** [comment]
I've been getting a lot of 413 errors because of this. Auto resizing or compressing the imager on the browser would be really useful!
